backend/src/main/java/codezap/template/service/TemplateService.java

```diff
@@ -136,27 +135,39 @@ public FindAllTemplatesResponse findAllBy(
         pageable = PageRequest.of(pageable.getPageNumber() - 1, pageable.getPageSize(), pageable.getSort());
 
         if (categoryId != null && tagIds != null) {
-            List<Long> templateIds = templateTagRepository.findAllTemplateIdInTagIds(tagIds, tagIds.size());
+            List<Long> templateIds = findTemplateIdContainsTagIds(tagIds);
+            categoryRepository.fetchById(categoryId);
             Page<Template> templatePage =
                     templateRepository.searchBy(memberId, keyword, categoryId, templateIds, pageable);
             return makeTemplatesResponseBy(templatePage);
         }
         if (categoryId != null) {
+            categoryRepository.fetchById(categoryId);
             Page<Template> templatePage = templateRepository.searchBy(memberId, keyword, categoryId, pageable);
             return makeTemplatesResponseBy(templatePage);
         }
         if (tagIds != null) {
-            List<Long> templateIds = templateTagRepository.findAllTemplateIdInTagIds(tagIds, tagIds.size());
+            List<Long> templateIds = findTemplateIdContainsTagIds(tagIds);
             Page<Template> templatePage = templateRepository.searchBy(memberId, keyword, templateIds, pageable);
             return makeTemplatesResponseBy(templatePage);
         }
         Page<Template> templatePage = templateRepository.searchBy(memberId, keyword, pageable);
         return makeTemplatesResponseBy(templatePage);
     }
 
+    private List<Long> findTemplateIdContainsTagIds(List<Long> tagIds) {
```

Comment:
Find는 반환값이 없을 수 있음을 암시해요. 
```suggestion
    private List<Long> fetchTemplateIdContainsTagIds(List<Long> tagIds) {
```
조회 결과가 없을 때 예외를 발생시키는 것을 의도했다면 fetch라고 명명하는 게 어떨까요?

Reply:
굿쟙입니다 ~~